src/components/src/main/java/org/apache/jmeter/extractor/gui/RegexExtractorGui.java

```diff
@@ -208,6 +214,8 @@ private JPanel makeParameterPanel() {
         templateField = new JLabeledTextField(JMeterUtils.getResString("template_field")); //$NON-NLS-1$
         refNameField = new JLabeledTextField(JMeterUtils.getResString("ref_name_field")); //$NON-NLS-1$
         matchNumberField = new JLabeledTextField(JMeterUtils.getResString("match_num_field")); //$NON-NLS-1$
+        failResultField = new JLabel(JMeterUtils.getResString("fail_if_not_matched_field"));
+        failResult = new JCheckBox();
```

Comment:
Could you use `JEditableCheckBox` here?

Reply:
I'm using JLabel + JCheckBox here because it's nicer to have the label in the same row as the others and the checkbox in the column of the input fields

Reply:
JEditableCheckbox would allow using `${..}` expressions while `JCheckBox`, see https://github.com/apache/jmeter/pull/5944

Reply:
I see, although the JEditableCheckbox is a nice feature to paramaterize scripts, I don't think it's a good usecase on this specific checkbox. I don't see a reason to parameterize a specific regex extractor. The other checkboxes on the Regex Extractor aren't an editable checkbox as well. 
I can give it a try tho, If it doesn't help, it doesn't hurt